### DISCUSSION TITLE
How to download full page source?

### DISCUSSION THREAD

**Discussion by 0x7FFFFFFFFFFFFFFF** [MontFerret/ferret#663] [Q&A] [answered]
I checked the examples and it seems all of them are scraping part of the webpage. How to download the full page source to a file from scraped URLs? Thanks.

**Comment by PierreBrisorgueil** [comment, marked as answer]
hello @0x7FFFFFFFFFFFFFFF,  just return your doc var  (you can play with INNET_HTML to select a part)

```
LET doc = DOCUMENT(CONCAT('https://duckduckgo.com/?q=', '%20', 'monferret'), {
    driver: 'cdp',
    viewport: {
        width: 1920,
        height: 1080
    }
})

RETURN doc
```

**Comment by 0x7FFFFFFFFFFFFFFF** [comment]
Thank you!
